### ISSUE TITLE
About Terabyte dataset preprocesing

### ISSUE THREAD

**Issue by BeyondCloud** [facebookresearch/dlrm#141]
Hi, 
I have preprocessed the dataset but I got the same issue as thread #126
The dimensions of processed emb_l input I got were [9979952, 36084, 17215 , 7376, 20137, 3, 7111, 1436, 61, 9757383, 1332440, 313978, 10,   2208 , 11177 , 129 , 4, 971, 14, 9994171, 7266769, 9946152,  415192, 12440, 102, 35], which is inconsistent with dlrm_emb64_subsample0.875_maxindrange10M_pretrained.pt.
Below was the command I used to run the code
```
python dlrm_s_pytorch.py 
--use-gpu 
--dataset-multiprocessing 
--data-sub-sample-rate=0.875 
--test-freq=10240 
--arch-embedding-size=9980333-36084-17217-7378-20134-3-7112-1442-61-9758201-1333352-313829-10-2208-11156-122-4-970-14-9994222-7267859-9946608-415421-12420-101-36 
--arch-mlp-bot="13-512-256-64" 
--arch-mlp-top="512-512-256-1" 
--load-model="./tb0875_10M.pt" 
--raw-data-file=./dataset/day 
--arch-sparse-feature-size=64 
--max-ind-range=10000000 
--mlperf-logging 
--memory-map 
--data-set=terabyte 
--data-generation="dataset" 
--round-targets=True 
--mini-batch-size=4 
--num-batches=100 
--print-freq=1 
--print-time
 --test-mini-batch-size=16384 
--test-num-workers=16 
```

**Comment by mnaumovfb** [comment]
As mentioned in the issue you have pointed out, the flags used for pre-processing data and the ones used to generate pre-trained models must match. If you compare your flags and the ones used by this command 
`./bench/dlrm_s_criteo_terabyte.sh ["--test-freq=10240 --memory-map --data-sub-sample-rate=0.875"]`
you will immediately see that they do not match. This is likely the cause of the mismatch in embedding table parameters.

Here is some of the differences between flags, where we denote [your_extra_flag] and {script_extra_flag}:
--use-gpu
[--dataset-multiprocessing] <-- very likely the culprit for the mismatch
--data-sub-sample-rate=0.875
--test-freq=10240
[--arch-embedding-size] <-- this is overwritten, so not a concern 
--arch-mlp-bot="13-512-256-64" 
--arch-mlp-top="512-512-256-1"
[--load-model="./tb0875_10M.pt"] <-- required for loading pre-saved model, so not a concern
--raw-data-file=./input/day 
{--processed-data-file=./input/terabyte_processed.npz} <-- should use a default value, but it's better to add it 
--arch-sparse-feature-size=64 
--max-ind-range=10000000
[--mlperf-logging] <-- changes the logging, but should not influence the model 
--memory-map
--data-set=terabyte  
--data-generation=dataset 
{--loss-function=bce} <-- will influence training and should be added 
--round-targets=True 
--mini-batch-size=2048 
{--learning-rate=0.1} <-- should use a default value, but it's better to add it
--print-freq=1024 
--print-time 
--test-mini-batch-size=16384 
--test-num-workers=16

This does not mean that your results are wrong, it's just that during pre-processing different elements could have been dropped from the datasets and this slightly changed the dictionaries and corresponding mapping to embedding tables. You either have to use the exact same flags or re-train the model.

**Comment by BeyondCloud** [comment]
Thank you very much! I tried your suggestion and it work perfectly.
I'll star your repo.
